test/e2e/specs/ExistingUserCreditCheckout.spec.js

```diff
@@ -0,0 +1,55 @@
+describe('Existing User Checkout with Kiva Credit', () => {
+	it('Baskets a loan, proceed to checkout, log in user with credit and purchases using kiva credit', () => {
+		let beginBalance;
+		let newBalance;
+		// get the test user id from cypress.env.json and set the cookie variable
+		const lenderID = Cypress.env().userInfo.kivaCreditLender.userID;
+		const loginCookie = `${lenderID}:recent/active/mfa`;
+
+		// Mock login our test lender
+		cy.setCookie('kvfa', loginCookie, 'domain=.kiva.org');
+		// go to portfolio and get current credit for later calculations
+		cy.visit('/portfolio');
+		cy.get('.strong .kiva-green').invoke('text').then(fullText => {
+			const number = fullText.match(/[0-9,.]+/);
+			beginBalance = parseFloat(number); // text to float
+		});
+		// Go to lend page
+		cy.visit('/lend?x');
+		// Set sort to random
+		cy.get('#filter-loan-sort-dropdown', { timeout: 10000 }).as('sortFilter');
+		cy.wait(5000);
```

Comment:
What is this particular wait needed for? Would it be possible to wait for an element to appear or request to return, depending on the situation?

Reply:
My guess is this is just to deal with slow loading pages in the lower envs...